src/include/mpitimpl.h

```diff
@@ -1238,5 +1241,48 @@ extern MPID_Thread_mutex_t mpi_t_mutex;
 /* Init and finalize routines */
 extern int MPIR_T_env_init(void);
 extern void MPIR_T_env_finalize(void);
+extern void MPIR_T_events_finalize(void);
+
+typedef MPI_Count(MPIR_T_timestamp_fn) (void);
+
+typedef struct MPIR_T_source_s {
+#ifdef HAVE_ERROR_CHECKING
+    MPIR_T_object_kind kind;
+#endif
+    int index;
+    char *name;
+    char *desc;
+    MPI_T_source_order ordering;
+    MPIR_T_timestamp_fn *timestamp_fn;
+    MPI_Count ticks_per_second;
+    MPI_Count max_ticks;
+
+    UT_hash_handle hh;          /* Makes this structure hashable */
+} MPIR_T_source_t;
```

Comment:
A newline between the struct and function declarations?

Reply:
Do we usually use `_t` suffix for `struct`s?  I usually associate `_t` types to some basic types such as `enums` or `integers`.

Reply:
We do in the tools code apparently 😄. I wasn't aware of the convention beforehand. Not strongly tied to it.